### DISCUSSION TITLE
I need creative ideas / help from drone experts

### DISCUSSION THREAD

**Discussion by joshgin98** [bitcraze/discussions#529] [Q&A]
Hello everyone! I hope you are all well, I need urgent help! I have already asked the question on reddit, unfortunately no one could help me there so far. I have attached the link for the picture of the obstacle course here: [https://preview.redd.it/qsueoe7ef7da1.png?width=1532&format=png&auto=webp&v=enabled&s=4cbfb2eda2b96f9111f5e736ec9606f66b0aacf5](url) , because unfortunately you can not post pictures in this forum.

i have been given the task to build a robot that overcomes this obstacle course and transports as many ping pong balls as possible to the target (5) (but one ball would also be ok). the robot should not exceed the dimensions 15x15x15 at the start, but may unfold/enlarge after the start. the robot should be programmed via arduino. the area under the net must not be entered by the robot, but the net can be used in any creative way (a grid that flexes a bit when loaded).

my current solution approach would be to build a quadcopter that simply: takes off - lands on the balls (using tape or some other construction to catch the balls on the ground) - takes off again and flies to the target area - lands in the target area.

However, it seems to me difficult to regulate the drone without remote control, does anyone have experience with this or another idea how it could work?

Or does anyone have a completely different idea how it could work easier?

thanks in advance :)

**Comment by joshgin98** [comment]
https://www.reddit.com/r/robotics/comments/10gxlwm/i_need_creative_robot_ideas/ link

**Comment by tobbeanton** [comment]
That is a difficult task. I guess you are not allowed to use any external sensors like cameras or positioning system? In that case it is very hard to do with the crazyflie ecosystem.

**Reply by joshgin98** [reply]
hi tobias! thanks for your answer. i am allowed to use sensors, cameras or position systems, the robot (or drone in my planned case) just must not be controlled remotely. i thought the crazyflie system might be good for this - but i am also grateful for any other hints how it might work. cheers!

**Reply by tobbeanton** [reply]
If you are allowed to use e.g. our [Lighthouse positioning system](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/lighthouse/) you would be able to control the drone precisely enough so it might be doable. I'm not sure the ping-ping ball would stay still when hovering above it though... You would also need en external camera to identify the ping pong balls. This would not be a beginners project though and one would need a lot of robotics experience to pull it off.

**Reply by knmcguire** [reply]
We did a christmas video of picking things up https://youtu.be/PUjlDpC7bVA.

But this required us to already know the location of the pingpong ball beforehand

**Comment by knmcguire** [comment]
Hi!

Sounds like a fun competition! Not sure how much we can help you as this is the forum of the [crazyflie platform](https://www.bitcraze.io/products/crazyflie-2-1/) and in terms of carrying things it will be limited.

So obstacle avoidance you could do with a Crazyflie + flowdeck + multiranger deck:
- https://www.bitcraze.io/documentation/tutorials/getting-started-with-stem-ranging-bundle/
- https://imrclab.github.io/crazyswarm2/tutorials.html#connecting-with-nav2-bringup

But carrying the pingpong ball will be difficult I think as that will obstruct the flow deck. And the weight of the mechanism might be too much already...

An alternative is to build something with the [autopilot board the Bolt ](https://www.bitcraze.io/products/crazyflie-bolt-1-1/), in combination with an Arduino. But I have to put a **big fat warning** there: Building your own drone, even as small as 15 cm, is difficult and potentially dangerous. And that holds even for remote controlling it, let alone letting it be autonomous. If you are inexperienced with robotic platforms, I would highly suggest starting with a driving robot. 

You can check out hackster.io for some example builds or on similar websites.

**Reply by joshgin98** [reply]
hi kimberly! thanks for your detailed answer. I was recommended the crazyflie system by a friend, but unfortunately he also is not quite familiar with it. I will have a look at the links provided, thank you and best regards!

**Reply by knmcguire** [reply]
Ah yes. But if you are also allowed to use external positioning systems then that might change thing though. I'll respond to tobias' comment
